python/lsst/ts/interlockmonitor/thermal/controller.py

```diff
@@ -0,0 +1,249 @@
+        self.master, self.slave = pty.openpty()
+        self.log.info("Create ports.")
+        self.log.info(f"{os.ttyname(self.master)}")
+        self.serial = aioserial.AioSerial(
+            port=os.ttyname(self.master), baudrate=19200, cancel_write_timeout=10
+        )
+        self.serial.fd = self.master
+        self.log.info("Open serial.")
+        return os.ttyname(self.slave)
+
+    async def reply(self):
+        """Generate reply and send data over port."""
+        while True:
+            for i in range(0, 5):
+                if i == 4:
+                    reply = f"C0{i+1}={self.temperatures[i]:03.4f}".encode("ISO-8859-1")
+                    bytes_written = await self.serial.write_async(reply)
+                    self.log.info(f"{bytes_written=}")
+                    bytes_written = await self.serial.write_async(
+                        aioserial.CR + aioserial.LF
+                    )
+                    self.log.info(f"{bytes_written=}")
+                else:
+                    reply = f"C0{i+1}={self.temperatures[i]:03.4f},".encode(
+                        "ISO-8859-1"
+                    )
+                    self.log.info(f"{reply=}")
+                    bytes_written = await self.serial.write_async(reply)
+                    self.log.info(f"{bytes_written=}")
+                await asyncio.sleep(0.175)
+
+    def close(self):
+        """Close the mock serial port."""
+        if self.serial is not None:
+            self.serial.close()
+            self.serial = None
+
+
+class ThermalScanner:
+    """Implement thermal scanner client.
+
+    Parameters
+    ----------
+    port : `str`
+        The device port.
+    baudrate : `int`
+        The device baudrate.
+
+    Attributes
+    ----------
+    port : `str`
+        The device port.
+    baudrate : `str`
+        The device baudrate.
+    serial : `aioserial.AioSerial`
+        The serial commander.
+    log : `logging.Logger`
+        The log instance.
+    """
+
+    def __init__(self, port, baudrate) -> None:
+        self.port = port
+        self.baudrate = baudrate
+        self.serial = aioserial.AioSerial(
+            port=self.port,
+            baudrate=self.baudrate,
+            parity=aioserial.PARITY_NONE,
+            stopbits=aioserial.STOPBITS_ONE,
+            bytesize=aioserial.EIGHTBITS,
+            cancel_read_timeout=2,
+        )
+        self.log = logging.getLogger(__name__)
+        self.log.info("Thermal Scanner client created.")
+
+    async def get_thermal_data(self):
+        """Get the thermal data from the device.
+
+        Returns
+        -------
+        thermal_data : `str`
+            The string representing thermal data from the scanner.
+        """
+        self.log.info("Reading thermal scanner.")
+        await asyncio.sleep(0.174 * 5)
+        self.thermal_data = await self.serial.read_until_async(aioserial.LF)
+        return self.thermal_data
+
+
+class ThermalScannerServer(tcpip.OneClientServer):
```

Comment:
This is too many classes in one module. Please separate out distinct classes into their own modules.

The mock and real thermal scanners _might_ belong together (I don't yet understand why you have both) but I think the server and mock client each should have their own module.

Please also use module names that better reflect the classes inside. "controller" doesn't seem to have a single class with Controller in its name.